beacon_node/network/src/network_beacon_processor/mod.rs

```diff
@@ -657,33 +657,26 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
             async move {
                 let slot_duration = beacon_chain.slot_clock.slot_duration();
                 let delay = beacon_chain.slot_clock.single_lookup_delay();
-                let interval_start = match (
-                    beacon_chain.slot_clock.duration_to_next_slot(),
-                    beacon_chain.slot_clock.seconds_from_current_slot_start(),
-                ) {
-                    (Some(duration_to_next_slot), Some(seconds_from_current_slot_start)) => {
-                        let duration_until_start = if seconds_from_current_slot_start > delay {
-                            duration_to_next_slot + delay
-                        } else {
-                            delay - seconds_from_current_slot_start
-                        };
-                        Instant::now() + duration_until_start
-                    }
-                    _ => {
-                        crit!(log,
-                        "Failed to read slot clock, delayed lookup service timing will be inaccurate.\
-                         This may degrade performance"
-                    );
-                        Instant::now()
-                    }
-                };
-
-                let mut interval = interval_at(interval_start, slot_duration);
                 loop {
-                    interval.tick().await;
+                    // Poll the delayed lookups for any block component requests
+                    // after the specified delay.
+                    // Keep remeasuring the offset rather than using an interval, so that we can correct
+                    // for system time clock adjustments.
+                    let wait = match beacon_chain.slot_clock.duration_to_next_slot() {
```

Comment:
there's an annoying edge case here where this is called after the start of a slot and before the delayed lookup instant. And then we'd end up sleeping for a full slot and then some instead of just waiting a second or two. I think it could only really happen on startup, but it also might be more likely we get a delayed lookup on startup cause we might have been offline during the gossip propagation of the block in the current slot.  The right behavior there is probably not to delay any lookups on the startup slot.. but I'm not sure how we'd do that either 🤔